transaction/core/src/fog_hint.rs

```diff
@@ -128,28 +101,25 @@ impl FogHint {
     /// # Arguments
     /// * ingest_server_private_key
     /// * encrypted fog hint payload
+    /// * default plaintext
+    /// * initialized output FogHint
     ///
     /// # Returns
-    /// * (Fog hint, true) on success (Default Fog hint, false) otherwise
+    /// * Choice(1) on success Choice(0) otherwise
+    #[inline(never)]
```

Comment:
Per [the documentation](https://doc.rust-lang.org/reference/attributes/codegen.html#the-inline-attribute): "`#[inline(never)]` *suggests* that an inline expansion should never be performed."

Unfortunately the more aggressive suggestion from chat---a conditional move of plaintext(s) based on the decrypt result (e.g. `ConditionallySelectable`) is the only option I'm aware of to make this actually constant-time.

Reply:
yeah `inline(never)` is not suitable for like establishing a security property

Reply:
Ok @jcape, I implemented ConditionallySelectable for a newtype wrapping the plaintext array, and did a conditional assign after the decrypt.  I think this was the main culprit.

I can also implement ConditionallySelectable on the FogHint itself, in case someone wants to use the returned Choice from ct_decrypt.  Do you want me to do that?